rest/config_new.go

```diff
@@ -0,0 +1,240 @@
+	ConfigUpdateFrequency time.Duration `json:"config_update_frequency,omitempty" help:"How often to poll Couchbase Server for new config changes. Default: 10s"`
+	Server                string        `json:"server,omitempty"                  help:"Couchbase Server connection string/URL"`
+	Username              string        `json:"username,omitempty"                help:"Username for authenticating to server"`
+	Password              string        `json:"password,omitempty"                help:"Password for authenticating to server"`
+	CertPath              string        `json:"cert_path,omitempty"               help:"Cert path (public key) for X.509 bucket auth"`
+	KeyPath               string        `json:"key_path,omitempty"                help:"Key path (private key) for X.509 bucket auth"`
+	CACertPath            string        `json:"ca_cert_path,omitempty"            help:"Root CA cert path for X.509 bucket auth"`
+}
+
+type APIConfig struct {
+	PublicInterface  string `json:"public_interface,omitempty"  help:"Network interface to bind public API to"`
+	AdminInterface   string `json:"admin_interface,omitempty"   help:"Network interface to bind admin API to"`
+	MetricsInterface string `json:"metrics_interface,omitempty" help:"Network interface to bind metrics API to"`
+	ProfileInterface string `json:"profile_interface,omitempty" help:"Network interface to bind profiling API to"`
+
+	ServerReadTimeout  time.Duration `json:"server_read_timeout,omitempty"  help:"maximum duration.Second before timing out read of the HTTP(S) request"`
+	ServerWriteTimeout time.Duration `json:"server_write_timeout,omitempty" help:"maximum duration.Second before timing out write of the HTTP(S) response"`
+	ReadHeaderTimeout  time.Duration `json:"read_header_timeout,omitempty"  help:"The amount of time allowed to read request headers"`
+	IdleTimeout        time.Duration `json:"idle_timeout,omitempty"         help:"The maximum amount of time to wait for the next request when keep-alives are enabled"`
+
+	Pretty             bool  `json:"pretty,omitempty"               help:"Pretty-print JSON responses"`
+	MaximumConnections uint  `json:"max_connections,omitempty"      help:"Max # of incoming HTTP connections to accept"`
+	CompressResponses  *bool `json:"compress_responses,omitempty"   help:"If false, disables compression of HTTP responses"`
+	HideProductVersion bool  `json:"hide_product_version,omitempty" help:"Whether product versions removed from Server headers and REST API responses"`
+
+	TLS  TLSConfig   `json:"tls,omitempty"`
+	CORS *CORSConfig `json:"cors,omitempty"`
+}
+
+type TLSConfig struct {
+	MinimumVersion string `json:"minimum_version,omitempty"     help:"The minimum allowable TLS version for the REST APIs"`
+	CertPath       string `json:"cert_path,omitempty" help:"The TLS cert file to use for the REST APIs"`
```

Comment:
Do you think that naming these properties the same as client certpath/keypath for connecting to Couchbase Server might lead to confusion, even if they are under a different hierarchy?

Reply:
I can see how they could be mixed up, although a misconfguration would be immediately obvious (either bootstrap auth fails, or invalid cert errors via HTTPS)

We could rename the Couchbase Server ones to be `x509_cert_path`/`x509_key_path` to make it super explicit what they're used for?

Reply:
The changes you've made look fine.